csrc/executor.cpp

```diff
@@ -179,6 +179,79 @@ void FusionExecutor::debugCompileFusionFromStr(
       fusion_id_ > 0, "assign a fusion_id_ <= 0 is not accepted.");
 }
 
+static inline DataType ComputeIndexType(
+    Fusion* fusion,
+    ExpressionEvaluator& expr_eval) {
+  FUSER_PERF_SCOPE("FusionExecutor::AllocOutputs");
+
+  std::vector<std::pair<int64_t, int64_t>> sizes_and_strides;
+
+  auto get_index_type = [&](TensorView* tv) {
```

Comment:
This doesn't work for non-contiguous tensors as nvfuser tensors don't have actual strides. What we could do would be for contiguous tensors we can just calculate the overall tensor size and use that to determine the index type. When non-contiguous tensors are detected, we would need to conservatively assume it may need 64-bit indexing.